Sources/SwiftProtobufCore/Message+BinaryAdditions.swift

```diff
@@ -16,18 +16,19 @@ import Foundation
 
 /// Binary encoding and decoding methods for messages.
 extension Message {
-  /// Returns a `Data` value containing the Protocol Buffer binary format
-  /// serialization of the message.
+  /// Returns a `SwiftProtobufContiguousBytes` instance containing the Protocol Buffer binary
+  /// format serialization of the message.
   ///
   /// - Parameters:
   ///   - partial: If `false` (the default), this method will check
   ///     `Message.isInitialized` before encoding to verify that all required
   ///     fields are present. If any are missing, this method throws
   ///     `BinaryEncodingError.missingRequiredFields`.
-  /// - Returns: A `Data` value containing the binary serialization of the
-  ///   message.
+  /// - Returns: A `SwiftProtobufContiguousBytes` instance containing the binary serialization
+  /// of the message.
+  ///
   /// - Throws: `BinaryEncodingError` if encoding fails.
-  public func serializedData(partial: Bool = false) throws -> Data {
+  public func serializedBytes<Bytes: SwiftProtobufContiguousBytes>(partial: Bool = false) throws -> Bytes {
```

Comment:
It may not be appropriate today, but I wonder if we should consider someday rephrasing this to _append_ the serialized message to a provided inout buffer object.   That would require a slightly different protocol with a way to obtain an initializing view.  (CC: @atrick)

Reply:
The client could do an append operation which provides an "initializing view" of the new storage, passing that to a closure. The closure would need to promise to initialize all the bytes. While you're holding onto a "view" of any kind, you won't be able to append or do anything that modifies the storage structure.

Reply:
But serializedDataSize() is internal... Having the message allocate the storage seems safer and more encapsulated.

Reply:
The only concern I might have with this approach is that it forces `SwiftProtobufContiguousBytes` to be able to dynamically allocate and manage storage. So this is more than just a "contiguous bytes" protocol now.
When we expose a public "contiguous bytes" protocol (a.k.a. BufferView), we can't make that guarantee.